RxBluetoothKit/BluetoothManager.swift

```diff
@@ -1,43 +1,74 @@
 import CoreBluetooth
 
+// swiftlint:disable line_length
+
 /**
- BluetoothManager is the main class which allows to use Bluetooth low energy to scan and connect to
- retrieved peripherals.
+ BluetoothManager is a class implementing ReactiveX API which wraps all Core Bluetooth Manager's functions allowing to
+ discover or connect to remote peripheral devices. It's using thin layer behind `RxCentralManagerType` protocol which is
+ implemented by `RxCBCentralManager` and should not be used directly by the user of a RxBluetoothKit library.
+
+ You can start using this class by discovering available services of nearby peripherals. Before calling any
+ public `BluetoothManager`'s functions you should make sure that Bluetooth is turned on and powered on. It can be done
+ by calling and observing returned value of `monitorState()` and then chaining it with `scanForPeripherals(_:options:)`:
+
+     bluetoothManager.monitorState()
+         .filter { $0 == .PoweredOn }
+         .take(1)
+         .flatMap { manager.scanForPeripherals(nil) }
+
+ As a result you will receive `ScannedPeripheral` which contains `Peripheral` object, `AdvertisementData` and
+ peripheral's RSSI registered during discovery. You can then `connectToPeripheral(_:options:)` and do other operations.
+
+ - seealso: `Peripheral`
  */
 public class BluetoothManager {
 
     /// Implementation of Central Manager
     private let centralManager: RxCentralManagerType
 
-    // Queue on which all stream are serialized
+    /// Queue on which all observables are serialised if needed
     private let subscriptionQueue: SerializedSubscriptionQueue
 
-    /// Internal structures lock
+    /// Lock which should be used before accessing any internal structures
     private let lock = NSLock()
 
-    // TODO: To be completed: seealso: scanForPeripherals
-    /// Queue of scan operations to be completed
+    /// Queue of scan operations which are waiting for an execution
     private var scanQueue: [ScanOperation] = []
-
     private let disposeBag = DisposeBag()
 
-    //MARK: Public interfaces
+    // MARK: Initialization
 
     /**
-     Create new Bluetooth Manager with specified implemention of Bluetooth Central Manager which will execute operations
-     by default on main thread.
+     Creates new `BluetoothManager` instance with specified implementation of `RxCentralManagerType` protocol which will be
+     used by this class. Most of a time `RxCBCentralManager` should be chosen by the user.
 
-     - Parameter centralManager: implementation of Central Manager
-     - Parameter queueScheduler: scheduler on which serialized work will be executed
+     - parameter centralManager: Implementation of `RxCentralManagerType` protocol used by this class.
+     - parameter queueScheduler: Scheduler on which all serialised operations are executed (such as scans). By default
+                                 main thread is used.
      */
     public init(centralManager: RxCentralManagerType,
```

Comment:
Think about making this method internal. And we need to talk about existence of SchedulerType